zingolib/src/wallet/transaction_records_by_id.rs

```diff
@@ -116,18 +116,14 @@ impl TransactionRecordsById {
     }
     /// Invalidates all transactions from a given height including the block with block height `reorg_height`
     ///
-    /// All information above a certain height is invalidated during a reorg.
+    /// All information above a certain height is invalidated during a reorg
     pub fn invalidate_all_transactions_after_or_at_height(&mut self, reorg_height: BlockHeight) {
         // First, collect txids that need to be removed
         let txids_to_remove = self
             .values()
             .filter_map(|transaction_metadata| {
-                if transaction_metadata
-                    .status
-                    .is_confirmed_after_or_at(&reorg_height)
-                    || transaction_metadata
-                        .status
-                        .is_pending_after_or_at(&reorg_height)
+                // doesnt matter the status: if it happen after a reorg, eliminate it
+                if transaction_metadata.status.get_height() >= reorg_height
```

Comment:
as the comment suggests below, we should only be invalidating confirmed txs. otherwise a re-org could cause Calculated / Transmitted but unsuccessful transactions to be discarded. anyway this is the same behaviour as before so this is non-blocking. and we havent implemented read/write for the cache anyway which would be best implemented by adding Transaction to TransactionRecord, lets leave this until its more relavent